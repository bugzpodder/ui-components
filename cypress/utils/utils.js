/* eslint-disable */
import uuid from "uuid";

export const getRandomCharacters = length => {
  return uuid
    .v4()
    .replace(/-/g, "")
    .substring(0, length);
};

export const generateRandomNumber = () => {
  return Array(5)
    .fill(0)
    .map(() => Math.random() * 1000 * 1000 * 1000)
    .map(Math.round)
    .map(value => `${value}`)
    .map(value => value.padStart(9, "0"))
    .join("");
};

export const getTextFromElement = element => {
  if (typeof element === "string") {
    return cy.get(element).then(getTextFromElement);
  }
  return cy.wrap(
    Cypress.$(element)
      .text()
      .trim(),
  );
};

export const getActiveSpinner = () => {
  return cy.get("[data-testid='spinner-overlay'][data-is-active='true']");
};

export const waitForProgressIndicator = () => {
  return getActiveSpinner().should("not.exist");
};
