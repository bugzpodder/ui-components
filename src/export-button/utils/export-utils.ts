import {
  ExportableColumn,
  ReportOptions,
  toDelimitedReport,
} from "@grailbio/lib";
import { saveAs } from "file-saver";

export const generateReport = (
  fileName: string,
  columns: ExportableColumn<any>[],
  data: Record<string, any>[],
  options: ReportOptions = {},
): void => {
  const output = toDelimitedReport(columns, data, options);
  const { fileMimeType = "text/csv" } = options;
  const blob = new Blob([output], { type: `${fileMimeType};charset=utf-8;` });
  try {
    saveAs(blob, fileName, true);
  } catch (error) {
    console.error("error saving blob", error);
  }
};
