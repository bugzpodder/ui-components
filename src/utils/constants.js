// @flow
export const COUNTRIES: Array<Object> = [
  { label: "Afghanistan" },
  { label: "Aland Islands" },
  { label: "Albania" },
  { label: "Algeria" },
  { label: "American Samoa" },
  { label: "Andorra" },
  { label: "Angola" },
  { label: "Anguilla" },
  { label: "Antarctica" },
  { label: "Argentina" },
  { label: "Armenia" },
  { label: "Aruba" },
  { label: "Australia" },
  { label: "Austria" },
  { label: "Azerbaijan" },
  { label: "Bahamas" },
  { label: "Bahrain" },
  { label: "Bangladesh" },
  { label: "Barbados" },
  { label: "Belarus" },
  { label: "Belgium" },
  { label: "Belize" },
  { label: "Benin" },
  { label: "Bermuda" },
  { label: "Bhutan" },
].map(suggestion => ({
  value: suggestion.label,
  label: suggestion.label,
}));
