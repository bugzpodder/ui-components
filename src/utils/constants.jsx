// @flow
import React from "react";

export const COUNTRIES: Array<Object> = [
  { label: "Afghanistan" },
  { label: "Aland Islands" },
  { label: "Albania" },
  { label: "Algeria" },
  { label: "American Samoa" },
  { label: "Andorra" },
  { label: "Angola" },
  { label: "Anguilla" },
  { label: "Antarctica" },
  { label: "Argentina" },
  { label: "Armenia" },
  { label: "Aruba" },
  { label: "Australia" },
  { label: "Austria" },
  { label: "Azerbaijan" },
  { label: "Bahamas" },
  { label: "Bahrain" },
  { label: "Bangladesh" },
  { label: "Barbados" },
  { label: "Belarus" },
  { label: "Belgium" },
  { label: "Belize" },
  { label: "Benin" },
  { label: "Bermuda" },
  { label: "Bhutan" },
].map(suggestion => ({
  label: suggestion.label,
  value: suggestion.label.toUpperCase().replace(" ", "_"),
}));

export const SERGEY =
  "https://media.licdn.com/dms/image/C5603AQGlBIPst1V4Ew/profile-displayphoto-shrink_800_800/0?e=1535587200&v=beta&t=2V2knyEhbTrhRWzgydsXLZqiX8gnoD1poj8jGeFpA0w";

export const EXAMPLE_TABLE_DATA = [
  { word: "Argle-bargle", origin: "Britain" },
  { word: "Dwaal", origin: "South Africa" },
  { word: "Floccinaucinihilipilification", origin: "General English" },
  { word: "Lollygag", origin: "North American" },
  { word: "Mouse Potato", origin: "General English" },
  { word: "Ogdoad", origin: "General English" },
  { word: "Sesquipedalian", origin: "General English" },
  { word: "Flibbertigibbet", origin: "Britain" },
  { word: "Curmudgeon", origin: "N/A" },
  { word: "Nudiustertian", origin: "Latin" },
  { word: "Tittynope", origin: "General English" },
  { word: "Pauciloquent", origin: "Britain" },
  { word: "Pneumonoultramicroscopicsilicovolcanoconiosis", origin: "General English" },
  { word: "Bijoux", origin: "Archaic" },
  { word: "Fartlek", origin: "Swedish" },
  { word: "Cruciverbalist", origin: "General English" },
  { word: "Erinaceous", origin: "General English" },
];

export const ADAMS_DATA = [
  {
    date: "1848-02-23T16:20:00Z",
    content: <div>John Quincy Adams dies.</div>,
  },
  {
    date: "1818-04-10T12:00:00Z",
    content: (
      <div>
        John Quincy Adams tries to dig to the middle of the Earth to conduct trade with the people living there.
      </div>
    ),
  },
  {
    date: "1767-07-11T12:20:00Z",
    content: <div>John Quincy Adams is born.</div>,
  },
];

export const EMU_DATA = [
  {
    date: "1932-12-10T16:20:00Z",
    content: <div>War ends and is declared a victory for the emus.</div>,
  },
  {
    date: "1932-11-08T12:00:00Z",
    content: <div>Over 2500 rounds had been fired with the media claiming that only a few emus had been killed.</div>,
  },
  {
    date: "1932-11-02T12:20:00Z",
    content: <div>Sir George Pearce wages war on the local population of emus in Australia.</div>,
  },
];
