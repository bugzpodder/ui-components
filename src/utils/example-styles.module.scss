.container {
  .spacing {
    margin-top: 10px;
    margin-right: 10px;
    margin-bottom: 20px;
    margin-left: 10px;
  }
}

.spinnerContainer {
  position: relative;

  .buttonContainer {
    display: inline-block;
      position: relative;
  }

  .spacing {
    margin-top: 10px;
    margin-right: 10px;
    margin-bottom: 20px;
    margin-left: 10px;
  }
}

