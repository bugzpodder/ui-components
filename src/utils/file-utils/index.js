// @flow
export { readFileListAsTextArray, readFileListAsText, readFileAsText } from "./file-utils";
