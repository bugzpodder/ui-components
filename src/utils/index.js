// @flow
export { COUNTRIES, SERGEY } from "./constants";
export { ExampleBlock } from "./example-block";
export { TestWrapper } from "./test-wrapper";
