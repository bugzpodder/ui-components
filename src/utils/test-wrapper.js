// @flow
import React, { type Node } from "react";
// $FlowFixMe: there is no createGenerateClassName export in @material-ui/core/styles
import { createGenerateClassName } from "@material-ui/core/styles";
// $FlowFixMe: Cannot resolve module react-jss/lib/JssProvider
import JssProvider from "react-jss/lib/JssProvider";
import "@grail/lib/src/client-utils/mocks";

type Props = {
	children: Node<*>,
};

const generateClassName = createGenerateClassName({
	dangerouslyUseGlobalCSS: true,
});

export const TestWrapper = (props: Props) => {
	const { children } = props;
	return <JssProvider generateClassName={generateClassName}>{children}</JssProvider>;
};
