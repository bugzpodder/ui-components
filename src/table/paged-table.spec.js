// @flow
import React from "react";
import { render, cleanup } from "react-testing-library";
import "jest-dom/extend-expect";
import mockConsole from "jest-mock-console";
import { TestWrapper } from "../utils";
import {
	columns,
	data,
	headerActions,
	invalidColumns,
	invalidTableOptions,
	tableOptions,
} from "./utilities/test-table-properties";
import { PagedTable } from "./index";

afterEach(cleanup);

/**
 Passing Cases
 **/

test("render simple paged table", () => {
	const { container } = render(
		<TestWrapper>
			<PagedTable
				columns={columns}
				data={data} />
		</TestWrapper>,
	);
	expect(container).toMatchSnapshot();
});

test("render paged table with no results", () => {
	const { container } = render(
		<TestWrapper>
			<PagedTable
				columns={columns}
				data={[]} />
		</TestWrapper>,
	);
	expect(container).toMatchSnapshot();
});

test("render loading paged table", () => {
	const { container } = render(
		<TestWrapper>
			<PagedTable
				columns={columns}
				data={data}
				isLoading={true} />
		</TestWrapper>,
	);
	expect(container).toMatchSnapshot();
});

test("render paged table with no results and is loading", () => {
	const { container } = render(
		<TestWrapper>
			<PagedTable
				columns={columns}
				data={[]}
				isLoading={true} />
		</TestWrapper>,
	);
	expect(container).toMatchSnapshot();
});

test("render paged table with no selected items", () => {
	const mockSelect = jest.fn();
	const { container } = render(
		<TestWrapper>
			<PagedTable
				columns={columns}
				data={data}
				onSelect={mockSelect}
				selectedRows={[]} />
		</TestWrapper>,
	);
	expect(container).toMatchSnapshot();
});

test("render full paged table", () => {
	const mockPagination = jest.fn();
	const mockSelect = jest.fn();
	const { container } = render(
		<TestWrapper>
			<PagedTable
				title="Test Table"
				idKey="columnOne"
				columns={columns}
				data={data}
				headerActions={headerActions}
				tableOptions={tableOptions}
				onPageChange={mockPagination}
				onSelect={mockSelect}
				selectedRows={tableOptions.selectedRowIds}
			/>
		</TestWrapper>,
	);
	expect(container).toMatchSnapshot();
});

/**
 Fail Cases
 **/

test("throw invalid paged table error", async () => {
	mockConsole();
	expect(() => render(<PagedTable columns={columns} />)).toThrowError();
	expect(console.error).toHaveBeenCalled();
});

test("throw invalid columns error", () => {
	mockConsole();
	expect(() => render(<PagedTable
		columns={invalidColumns}
		data={data} />)).toThrowError();
	expect(console.error).toHaveBeenCalled();
});

test("throw invalid pagination prop error", () => {
	const mockPagination = jest.fn();
	mockConsole();
	expect(() => render(<PagedTable
		columns={columns}
		data={data}
		onPageChange={mockPagination} />)).toThrowError();
	expect(console.error).toHaveBeenCalled();
});

test("throw invalid pagination options error", () => {
	mockConsole();
	expect(() =>
		render(<PagedTable
			columns={columns}
			data={data}
			onPageChange={() => {}}
			tableOptions={invalidTableOptions} />),
	).toThrowError();
	expect(console.error).toHaveBeenCalled();
});

test("throw invalid sortOptions error", () => {
	mockConsole();
	expect(() => render(<PagedTable
		columns={columns}
		data={data}
		onSort={() => {}} />)).toThrowError();
	expect(console.error).toHaveBeenCalled();
});
