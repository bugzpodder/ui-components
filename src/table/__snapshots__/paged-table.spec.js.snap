// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`render full paged table 1`] = `
<div>
  <div
    class="tableContainer"
  >
    <div
      class="MuiPaper-root MuiPaper-elevation6 MuiPaper-rounded MuiCard-root tableCard PagedTable card"
    >
      <div
        class="MuiCardHeader-root"
        data-testid="card-header"
      >
        <div
          class="MuiCardHeader-content"
        >
          <span
            class="MuiTypography-root MuiTypography-headline MuiCardHeader-title card-title"
          >
            Test Table (1 Selected)
          </span>
          
        </div>
        <div
          class="MuiCardHeader-action headerActions"
        >
          <p
            class="MuiTypography-root MuiTypography-body1"
          >
            Test Header Action
          </p>
        </div>
      </div>
      <div
        class="MuiCardContent-root tableContent"
        data-testid="card-content"
      >
        <table
          class="MuiTable-root"
        >
          <thead
            class="MuiTableHead-root TableHeader"
          >
            <tr
              class="MuiTableRow-root MuiTableRow-head"
            >
              <th
                class="MuiTableCell-root MuiTableCell-head COLUMN_SELECT-header singleIcon"
                scope="col"
              >
                <span
                  class="MuiButtonBase-root MuiIconButton-root MuiSwitchBase-root MuiCheckbox-root MuiCheckbox-colorPrimary MuiSwitchBase-checked MuiCheckbox-checked active indeterminate"
                  data-testid="paged-table-checkbox-header"
                >
                  <span
                    class="MuiIconButton-label"
                  >
                    <svg
                      aria-hidden="true"
                      class="MuiSvgIcon-root"
                      focusable="false"
                      viewBox="0 0 24 24"
                    >
                      <path
                        d="M19 3H5c-1.11 0-2 .9-2 2v14c0 1.1.89 2 2 2h14c1.11 0 2-.9 2-2V5c0-1.1-.89-2-2-2zm-9 14l-5-5 1.41-1.41L10 14.17l7.59-7.59L19 8l-9 9z"
                      />
                    </svg>
                    <input
                      class="MuiSwitchBase-input"
                      type="checkbox"
                      value=""
                    />
                  </span>
                  <span
                    class="MuiTouchRipple-root"
                  />
                </span>
              </th>
              <th
                class="MuiTableCell-root MuiTableCell-head columnOne-header tableHeader"
                scope="col"
              >
                Column One
              </th>
              <th
                class="MuiTableCell-root MuiTableCell-head columnThree-header tableHeader"
                scope="col"
              >
                Column Two
              </th>
            </tr>
          </thead>
          <tbody
            class="MuiTableBody-root"
          >
            <tr
              class="MuiTableRow-root"
              data-testid="First Datum"
            >
              <td
                class="MuiTableCell-root MuiTableCell-body tableCell singleIcon singleIcon"
                data-cell-id="First Datum-0"
              >
                <span
                  class="MuiButtonBase-root MuiIconButton-root MuiSwitchBase-root MuiCheckbox-root MuiCheckbox-colorPrimary MuiSwitchBase-checked MuiCheckbox-checked active"
                  data-testid="paged-table-checkbox-cell"
                >
                  <span
                    class="MuiIconButton-label"
                  >
                    <svg
                      aria-hidden="true"
                      class="MuiSvgIcon-root"
                      focusable="false"
                      viewBox="0 0 24 24"
                    >
                      <path
                        d="M19 3H5c-1.11 0-2 .9-2 2v14c0 1.1.89 2 2 2h14c1.11 0 2-.9 2-2V5c0-1.1-.89-2-2-2zm-9 14l-5-5 1.41-1.41L10 14.17l7.59-7.59L19 8l-9 9z"
                      />
                    </svg>
                    <input
                      class="MuiSwitchBase-input"
                      id="First Datum"
                      type="checkbox"
                      value=""
                    />
                  </span>
                  <span
                    class="MuiTouchRipple-root"
                  />
                </span>
              </td>
              <td
                class="MuiTableCell-root MuiTableCell-body tableCell"
                data-cell-id="First Datum-1"
              >
                <span>
                  First Datum
                </span>
              </td>
              <td
                class="MuiTableCell-root MuiTableCell-body tableCell"
                data-cell-id="First Datum-2"
              />
            </tr>
          </tbody>
        </table>
      </div>
      <div
        class="MuiCardActions-root tableFooter"
        data-testid="card-actions"
      >
        <div
          class="table-pager tablePager"
        >
          <div
            class="MuiToolbar-root MuiToolbar-gutters MuiTablePagination-toolbar"
          >
            <div
              class="MuiTablePagination-spacer"
            />
            <span
              class="MuiTypography-root MuiTypography-caption MuiTablePagination-caption"
            >
              Rows per page:
            </span>
            <div
              class="MuiInput-root select-row-count selectRowCount"
            >
              <div
                class="MuiSelect-root MuiTablePagination-selectRoot"
              >
                <div
                  aria-haspopup="true"
                  aria-pressed="false"
                  class="MuiSelect-select MuiTablePagination-select MuiSelect-selectMenu MuiInput-input"
                  role="button"
                  tabindex="0"
                >
                  5
                </div>
                <input
                  id="rows-per-page"
                  type="hidden"
                  value="5"
                />
                <svg
                  aria-hidden="true"
                  class="MuiSvgIcon-root MuiSelect-icon MuiTablePagination-selectIcon"
                  focusable="false"
                  viewBox="0 0 24 24"
                >
                  <path
                    d="M7 10l5 5 5-5z"
                  />
                </svg>
              </div>
            </div>
            <span
              class="MuiTypography-root MuiTypography-caption MuiTablePagination-caption"
            >
              Showing 1 to 1
            </span>
            <div
              class="MuiTablePagination-actions"
            >
              <button
                class="MuiButtonBase-root MuiButtonBase-disabled MuiIconButton-root MuiIconButton-disabled previous-page"
                disabled=""
                tabindex="-1"
                type="button"
              >
                <span
                  class="MuiIconButton-label"
                >
                  <svg
                    aria-hidden="true"
                    class="MuiSvgIcon-root"
                    focusable="false"
                    viewBox="0 0 24 24"
                  >
                    <path
                      d="M15.41 16.09l-4.58-4.59 4.58-4.59L14 5.5l-6 6 6 6z"
                    />
                  </svg>
                </span>
              </button>
              <button
                class="MuiButtonBase-root MuiButtonBase-disabled MuiIconButton-root MuiIconButton-disabled next-page"
                disabled=""
                tabindex="-1"
                type="button"
              >
                <span
                  class="MuiIconButton-label"
                >
                  <svg
                    aria-hidden="true"
                    class="MuiSvgIcon-root"
                    focusable="false"
                    viewBox="0 0 24 24"
                  >
                    <path
                      d="M8.59 16.34l4.58-4.59-4.58-4.59L10 5.75l6 6-6 6z"
                    />
                  </svg>
                </span>
              </button>
            </div>
          </div>
        </div>
      </div>
    </div>
  </div>
</div>
`;

exports[`render loading paged table 1`] = `
<div>
  <div
    class="tableContainer"
  >
    <div
      class="MuiPaper-root MuiPaper-elevation6 MuiPaper-rounded MuiCard-root tableCard PagedTable card"
    >
      <div
        class="MuiCardContent-root tableContent"
        data-testid="card-content"
      >
        <table
          class="MuiTable-root"
        >
          <thead
            class="MuiTableHead-root TableHeader"
          >
            <tr
              class="MuiTableRow-root MuiTableRow-head"
            >
              <th
                class="MuiTableCell-root MuiTableCell-head columnOne-header tableHeader"
                scope="col"
              >
                Column One
              </th>
              <th
                class="MuiTableCell-root MuiTableCell-head columnThree-header tableHeader"
                scope="col"
              >
                Column Two
              </th>
            </tr>
          </thead>
          <tbody
            class="MuiTableBody-root"
          >
            <tr
              class="MuiTableRow-root"
              data-testid="0"
            >
              <td
                class="MuiTableCell-root MuiTableCell-body tableCell"
                data-cell-id="0-0"
              >
                <span>
                  First Datum
                </span>
              </td>
              <td
                class="MuiTableCell-root MuiTableCell-body tableCell"
                data-cell-id="0-1"
              />
            </tr>
          </tbody>
        </table>
      </div>
    </div>
    <div
      class="spinnerOverlay active"
      data-is-active="true"
      data-testid="spinner-overlay"
    >
      <div
        class="loading"
      />
    </div>
  </div>
</div>
`;

exports[`render paged table with no results 1`] = `
<div>
  <div
    class="tableContainer"
  >
    <div
      class="MuiPaper-root MuiPaper-elevation6 MuiPaper-rounded MuiCard-root tableCard PagedTable card"
    >
      <div
        class="MuiCardContent-root tableContent"
        data-testid="card-content"
      >
        <table
          class="MuiTable-root"
        >
          <thead
            class="MuiTableHead-root TableHeader"
          >
            <tr
              class="MuiTableRow-root MuiTableRow-head"
            >
              <th
                class="MuiTableCell-root MuiTableCell-head columnOne-header tableHeader"
                scope="col"
              >
                Column One
              </th>
              <th
                class="MuiTableCell-root MuiTableCell-head columnThree-header tableHeader"
                scope="col"
              >
                Column Two
              </th>
            </tr>
          </thead>
          <tbody
            class="MuiTableBody-root"
          >
            <tr
              class="MuiTableRow-root"
            >
              <td
                class="MuiTableCell-root MuiTableCell-body no-results noResults"
                colspan="2"
              >
                0 Results
              </td>
            </tr>
          </tbody>
        </table>
      </div>
    </div>
  </div>
</div>
`;

exports[`render paged table with no results and is loading 1`] = `
<div>
  <div
    class="tableContainer"
  >
    <div
      class="MuiPaper-root MuiPaper-elevation6 MuiPaper-rounded MuiCard-root tableCard PagedTable card"
    >
      <div
        class="MuiCardContent-root tableContent"
        data-testid="card-content"
      >
        <table
          class="MuiTable-root"
        >
          <thead
            class="MuiTableHead-root TableHeader"
          >
            <tr
              class="MuiTableRow-root MuiTableRow-head"
            >
              <th
                class="MuiTableCell-root MuiTableCell-head columnOne-header tableHeader"
                scope="col"
              >
                Column One
              </th>
              <th
                class="MuiTableCell-root MuiTableCell-head columnThree-header tableHeader"
                scope="col"
              >
                Column Two
              </th>
            </tr>
          </thead>
          <tbody
            class="MuiTableBody-root"
          >
            <tr
              class="MuiTableRow-root"
            >
              <td
                class="MuiTableCell-root MuiTableCell-body no-results noResults"
                colspan="2"
              />
            </tr>
          </tbody>
        </table>
      </div>
    </div>
    <div
      class="spinnerOverlay active"
      data-is-active="true"
      data-testid="spinner-overlay"
    >
      <div
        class="loading"
      />
    </div>
  </div>
</div>
`;

exports[`render paged table with no selected items 1`] = `
<div>
  <div
    class="tableContainer"
  >
    <div
      class="MuiPaper-root MuiPaper-elevation6 MuiPaper-rounded MuiCard-root tableCard PagedTable card"
    >
      <div
        class="MuiCardContent-root tableContent"
        data-testid="card-content"
      >
        <table
          class="MuiTable-root"
        >
          <thead
            class="MuiTableHead-root TableHeader"
          >
            <tr
              class="MuiTableRow-root MuiTableRow-head"
            >
              <th
                class="MuiTableCell-root MuiTableCell-head COLUMN_SELECT-header singleIcon"
                scope="col"
              >
                <span
                  class="MuiButtonBase-root MuiIconButton-root MuiSwitchBase-root MuiCheckbox-root MuiCheckbox-colorPrimary"
                  data-testid="paged-table-checkbox-header"
                >
                  <span
                    class="MuiIconButton-label"
                  >
                    <svg
                      aria-hidden="true"
                      class="MuiSvgIcon-root"
                      focusable="false"
                      viewBox="0 0 24 24"
                    >
                      <path
                        d="M19 5v14H5V5h14m0-2H5c-1.1 0-2 .9-2 2v14c0 1.1.9 2 2 2h14c1.1 0 2-.9 2-2V5c0-1.1-.9-2-2-2z"
                      />
                    </svg>
                    <input
                      class="MuiSwitchBase-input"
                      type="checkbox"
                      value=""
                    />
                  </span>
                  <span
                    class="MuiTouchRipple-root"
                  />
                </span>
              </th>
              <th
                class="MuiTableCell-root MuiTableCell-head columnOne-header tableHeader"
                scope="col"
              >
                Column One
              </th>
              <th
                class="MuiTableCell-root MuiTableCell-head columnThree-header tableHeader"
                scope="col"
              >
                Column Two
              </th>
            </tr>
          </thead>
          <tbody
            class="MuiTableBody-root"
          >
            <tr
              class="MuiTableRow-root"
              data-testid="0"
            >
              <td
                class="MuiTableCell-root MuiTableCell-body tableCell singleIcon singleIcon"
                data-cell-id="0-0"
              >
                <span
                  class="MuiButtonBase-root MuiIconButton-root MuiSwitchBase-root MuiCheckbox-root MuiCheckbox-colorPrimary"
                  data-testid="paged-table-checkbox-cell"
                >
                  <span
                    class="MuiIconButton-label"
                  >
                    <svg
                      aria-hidden="true"
                      class="MuiSvgIcon-root"
                      focusable="false"
                      viewBox="0 0 24 24"
                    >
                      <path
                        d="M19 5v14H5V5h14m0-2H5c-1.1 0-2 .9-2 2v14c0 1.1.9 2 2 2h14c1.1 0 2-.9 2-2V5c0-1.1-.9-2-2-2z"
                      />
                    </svg>
                    <input
                      class="MuiSwitchBase-input"
                      id="0"
                      type="checkbox"
                      value=""
                    />
                  </span>
                  <span
                    class="MuiTouchRipple-root"
                  />
                </span>
              </td>
              <td
                class="MuiTableCell-root MuiTableCell-body tableCell"
                data-cell-id="0-1"
              >
                <span>
                  First Datum
                </span>
              </td>
              <td
                class="MuiTableCell-root MuiTableCell-body tableCell"
                data-cell-id="0-2"
              />
            </tr>
          </tbody>
        </table>
      </div>
    </div>
  </div>
</div>
`;

exports[`render simple paged table 1`] = `
<div>
  <div
    class="tableContainer"
  >
    <div
      class="MuiPaper-root MuiPaper-elevation6 MuiPaper-rounded MuiCard-root tableCard PagedTable card"
    >
      <div
        class="MuiCardContent-root tableContent"
        data-testid="card-content"
      >
        <table
          class="MuiTable-root"
        >
          <thead
            class="MuiTableHead-root TableHeader"
          >
            <tr
              class="MuiTableRow-root MuiTableRow-head"
            >
              <th
                class="MuiTableCell-root MuiTableCell-head columnOne-header tableHeader"
                scope="col"
              >
                Column One
              </th>
              <th
                class="MuiTableCell-root MuiTableCell-head columnThree-header tableHeader"
                scope="col"
              >
                Column Two
              </th>
            </tr>
          </thead>
          <tbody
            class="MuiTableBody-root"
          >
            <tr
              class="MuiTableRow-root"
              data-testid="0"
            >
              <td
                class="MuiTableCell-root MuiTableCell-body tableCell"
                data-cell-id="0-0"
              >
                <span>
                  First Datum
                </span>
              </td>
              <td
                class="MuiTableCell-root MuiTableCell-body tableCell"
                data-cell-id="0-1"
              />
            </tr>
          </tbody>
        </table>
      </div>
    </div>
  </div>
</div>
`;
